AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -3189,13 +3204,28 @@ - (void)setHierarchyState:(ASHierarchyState)newState
       // Entering layout pending state
     } else {
       // Leaving layout pending state, reset related properties
-      {
-        ASDN::MutexLocker l(__instanceLock__);
-        _pendingTransitionID = ASLayoutElementContextInvalidTransitionID;
-        _pendingLayoutTransition = nil;
+      ASDN::MutexLocker l(__instanceLock__);
+      _pendingTransitionID = ASLayoutElementContextInvalidTransitionID;
+      _pendingLayoutTransition = nil;
+    }
+  }
+
+#if YOGA /* YOGA */
```

Comment:
The hierarchyState integration is the biggest, but it is very simple code that follows a pattern of the other if() cases